test/textbringer/commands/test_buffers.rb

```diff
@@ -323,4 +323,34 @@ def test_read_only_mode
     assert_equal(true, Buffer.current.read_only?)
     assert_equal(true, Buffer.current.modified?)
   end
+
+  def test_kill_region_on_read_only_buffer
+    buffer = Buffer.current
+    insert("hello world")
+    buffer.read_only = true
+    
+    # Select "world"
+    goto_char(6)
+    set_mark_command
+    end_of_buffer
+
+    message_received = nil
+    echo_area = Window.echo_area
+    
+    # Temporarily override the show method of the echo_area for this test
+    echo_area.singleton_class.define_method(:show) do |text|
+      message_received = text
+    end
+
+    begin
+      kill_region
+    ensure
+      # Restore the original method to avoid side effects
+      echo_area.singleton_class.remove_method(:show)
+    end
```

Comment:
Please don't override Window.echo_area.show in tests.
You can't get the displayed message by `Window.echo_area.message`.

However, you don't need to use it if clipboard_kill_region is fixed as I stated above.